napari/layers/utils/color_manager.py

```diff
@@ -142,6 +142,12 @@ class ColorManager(EventedModel):
         The colors in a Nx4 color array, where N is the number of colors.
     """
 
+    class Config:
+        # override EventedModel which defaults to 2 (inplace mutation)
+        # note that if we wanted some fields to have inplace mutations and some not,
+        # we would still have to set this to 1 or the global setting would win
+        allow_mutation = 1
+
```

Comment:
This allows us to the retain the current behaviour of fields that accepts different types among which one or more are evented. Just like now, this will happily replace an evented object, breaking sources and event callbacks.

Actually, we could set `allow_mutation = 1` to individual fields, which is probably better.